power_grid_model_c/power_grid_model/include/power_grid_model/common/exception.hpp

```diff
@@ -112,7 +112,12 @@ class SparseMatrixError : public PowerGridError {
 
 class NotObservableError : public PowerGridError {
   public:
-    NotObservableError() { append_msg("Not enough measurements available for state estimation.\n"); }
+    NotObservableError(std::string const& msg = "") {
```

Comment:
might be good to double-check that the python exception wrapper still works as intended

Reply:
Message is appended to the back. re would still match.